lib/rets/metadata/file_cache.rb

```diff
@@ -0,0 +1,29 @@
+module Rets
+  module Metadata
+
+    # This metadata cache persists the metadata to a file.
+    class FileCache
+
+      def initialize(path)
+        @path = path
+      end
+
+      # Save the metadata.
+      def save(metadata)
+        File.open(@path, "wb") do |file|
+          Marshal.dump(metadata, file)
+        end
+      end
+
+      # Load the metadata.  Return nil if it could not be loaded.
+      def load
+        File.open(@path, "rb") do |file|
+          Marshal.load(file)
+        end
+      rescue IOError, SystemCallError
+        nil
+      end
```

Comment:
We're doing exactly that, since the root object has a marshal_dump that only saves the sources:

https://github.com/wconrad/rets/blob/easier-metadata-caching/lib/rets/metadata/root.rb#L65

Then the (new) marshal_load just recovers the source:

https://github.com/wconrad/rets/blob/easier-metadata-caching/lib/rets/metadata/root.rb#L69

The log is set by the Client, after the cache has done its thing:

https://github.com/wconrad/rets/blob/easier-metadata-caching/lib/rets/client.rb#L340

In any case, whether we pass the root object or the source to the cache object, I like the idea that the author of the cache should treat the object as some random object to be marshalled/unmarshalled.  Let's keep decisions about what needs to be cached in the hands of the library, and just rely on the user to do the actual save/restore.